go.mod

```diff
@@ -1,57 +1,84 @@
 module sigs.k8s.io/aws-iam-authenticator
 
-go 1.16
```

Comment:
this defines lowest go version. will user have to upgrade from 1.16/1.17/1.18?

Reply:
I think we should upgrade the version (at least to the last supported version by the go team, which is 1.18 right now I believe) but I'll do it in a separate PR